### PR TITLE
file system cache doc change

### PR DESCRIPTION
**Pull request by MehulBatra**
## Description
File system cache document change on optional fields

## Additional context and related issues
As a developer when we try to enable caching, we face errors for setting options and try to look through it via logs what went wrong, if we could specify proactively that would make things easier.

## Release notes

(*) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: docs/src/main/sphinx/object-storage/file-system-cache.md [hunk 1]
```diff
@@ -112,16 +112,14 @@ enable and configure caching for the specific catalogs.
     and set the values for `fs.cache.max-sizes` or
     `fs.cache.max-disk-usage-percentages` accordingly.
 * - `fs.cache.max-sizes`
-  - Optional, comma-separated list of maximum [data sizes](prop-type-data-size)
-    for each caching directory. Order of values must be identical to the
-    directories list. Can not be used together with
-    `fs.cache.max-disk-usage-percentages`.
+  - (Optional), comma-separated list specifying the maximum [data sizes](prop-type-data-size)
+    for each caching directory.The order of values must correspond to the order of the directories list. 
+    This option cannot be used in conjunction with `fs.cache.max-disk-usage-percentages`.
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/file-system-cache.md:117]: Wording should be kept simple .. 

```suggestion
    This property cannot be used with `fs.cache.max-disk-usage-percentages`.
```

File: docs/src/main/sphinx/object-storage/file-system-cache.md [hunk 2]
```diff
@@ -112,16 +112,14 @@ enable and configure caching for the specific catalogs.
...
 * - `fs.cache.max-disk-usage-percentages`
-  - Optional, comma-separated list of maximum percentage values of the used disk
-    for each directory. Each value is an integer between 1 and 100. Order of
-    values must be identical to the directories list. If multiple directories
-    use the same disk, ensure that total percentages per drive remains below 100
-    percent. Can not be used together with `fs.cache.max-sizes`.
+  - (Optional), comma-separated list indicating the maximum percentage values of disk usage for each directory. 
+    Each value must be an integer between 1 and 100, and the order must match the directories list.
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/file-system-cache.md:120]:
```suggestion
    Each value must be an integer between 1 and 100. The order of values must be identical to the directories list.
```

File: docs/src/main/sphinx/object-storage/file-system-cache.md [hunk 3]
```diff
@@ -112,16 +112,14 @@ enable and configure caching for the specific catalogs.
...
+    If multiple directories share the same disk, ensure the total percentages for that drive do not exceed 100%.
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/file-system-cache.md:121]:
```suggestion
    If multiple directories share the same disk, ensure that the total percentages for that drive do not exceed 100%.
```

File: docs/src/main/sphinx/object-storage/file-system-cache.md [hunk 4]
```diff
@@ -112,16 +112,14 @@ enable and configure caching for the specific catalogs.
...
+    This option cannot be used in conjunction with `fs.cache.max-sizes`.
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/file-system-cache.md:122]:
```suggestion
    This property cannot be used with `fs.cache.max-sizes`.
```

**Review by mosabua — CHANGES_REQUESTED** [review]
Not sure this improves much and need to figure out if or how we highlight optional vs required. The issue is that we do NOT do that anywhere else in the docs and it really should stay consistent. Also the braces are definitely against the Google doc style we use, italics might be fine but still seems against the rest of the docs and overkill.

**Comment by mosabua** [comment]
Also update commit message to follow https://cbea.ms/git-commit/

File: docs/src/main/sphinx/object-storage/file-system-cache.md (hunk 4 above)
Comment (colebow) [inline docs/src/main/sphinx/object-storage/file-system-cache.md:122]:
```suggestion
  - Comma-separated list indicating the maximum percentage values of disk usage for each directory. 
    Each value must be an integer between 1 and 100, and the order must match the directories list.
    If multiple directories share the same disk, ensure the total percentages for that drive do not exceed 100%.
    This option cannot be used with `fs.cache.max-sizes`, though one must be set.
```

Comment (colebow) [reply docs/src/main/sphinx/object-storage/file-system-cache.md:122]: I think I'd prefer this verbiage, it's a little more wordy, but it's way more clear (and apply this to the above property as well). We need to make it obvious that you have to set one of these two properties.

**Review by colebow — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/object-storage/file-system-cache.md (hunk 4 above)
Comment (mosabua) [reply docs/src/main/sphinx/object-storage/file-system-cache.md:122]: Thats fine as well with me .. not sure we should drop the initial "Optional"

**Comment by MehulBatra** [comment]
> I chatted with some other writers on the team - for now we do not want to use () or italics for the required and optional terms since it does not provide much additional benefit but introduces an inconsistency with all the other docs.

I picked up the idea of () from referring to the Trino Kubernetes docs itself, I found it quite clear as it was explicitly marked optional and caught my attention. 
image

File: docs/src/main/sphinx/object-storage/file-system-cache.md [hunk 5]
```diff
@@ -102,7 +102,7 @@ enable and configure caching for the specific catalogs.
 * - `fs.cache.enabled`
   - Enable object storage caching. Defaults to no caching with the value `false`.
 * - `fs.cache.directories`
-  - Required, comma-separated list of absolute paths to directories to use for
+  - (Required), comma-separated list of absolute paths to directories to use for
```
[the remark this answers is not in the document]
Comment (MehulBatra) [reply docs/src/main/sphinx/object-storage/file-system-cache.md:105]: Thanks for the feedback I will refer to that from now on for updates!

**Comment by mosabua** [comment]
> > I chatted with some other writers on the team - for now we do not want to use () or italics for the required and optional terms since it does not provide much additional benefit but introduces an inconsistency with all the other docs.
> 
> @mosabua I picked up the idea of () from referring to the Trino Kubernetes docs itself, I found it quite clear as it was explicitly marked optional and caught my attention. image

Hm .. interesting .. I am not sure but to keep it simple I would suggest not to do that for now .. we can revisit for the complete docs or maybe take it out in the k8s docs.

**Comment by mosabua** [comment]
Also please squash commits and update commit message and submit CLA.

**Comment by MehulBatra** [comment]
> Also please squash commits and update commit message and submit CLA.

CLA I have already submitted via email, changing the commit message and squashing it!

**Comment by mosabua** [comment]
Please fix wrapping to 80 character columns while we wait for CLA processing

**Review by mosabua — APPROVED** [review]
Looks good now. Just waiting for CLA processing to be done by @martint

**Comment by MehulBatra** [comment]
@mosabua @martint when can we expect the cla to be processed, thank you!
Please do let me know if anything is missing from my end.

**Comment by mosabua** [comment]
@cla-bot check

**Comment by mosabua** [comment]
I just recently fixed something else in this section so you will have to rebse @MehulBatra .. sorry.

**Comment by mosabua** [comment]
@cla-bot check

**Comment by mosabua** [comment]
@MehulBatra CLA was processed now, seems like your git or github config is not using the right email and CLA bot verification is not working, also please rebase and resolve conflict. 

Once all is good please ping me and I will review again and merge.

**Comment by MehulBatra** [comment]
I am still working on it to get CLA resolved

**Comment by mosabua** [comment]
Please

**Comment by mosabua** [comment]
I think all concerns are already addressed .. if not please reopen this PR and rebase, or send a new PR.

**Pull request closed without merging**